src/config/mod.rs

```diff
@@ -1157,7 +1224,28 @@ impl DbConfig {
         }
     }
 
-    pub fn build_opt(&self, stats: Option<&RocksStatistics>) -> RocksDbOptions {
+    pub fn build_shared(&self, env: Arc<Env>) -> SharedBetweenDbs {
+        SharedBetweenDbs {
+            env,
+            statistics: Arc::new(RocksStatistics::new_titan()),
+            rate_limiter: Some(Arc::new(RateLimiter::new_writeampbased_with_auto_tuned(
```

Comment:
What if `rate_bytes_per_sec` is set to 0 or `rate_limiter_auto_tuned` is set to false?

Reply:
It's used regardlessly so that it can always be dynamically enabled later. Behavior for the disabled case is the same as native rate limiter.

Reply:
This is a behavior change to v1 that I'm not sure whether is appropriate. For example, when auto_tuned is disable, a different rate limiter is used. I suggest to leave "dynamically enabled" as a different feature instead.

Reply:
The old code is already dynamically changeable. I.e. the code path used when starting with auto-tuned=false is different from starting with auto-tuned=true then dynamically disable it.